src/Cli/Microsoft.TemplateEngine.Cli/Commands/Extensions.cs

```diff
@@ -25,24 +26,15 @@ internal static class Extensions
         /// <summary>
         /// Checks if <paramref name="parseResult"/> contains an error for <paramref name="option"/>.
         /// </summary>
-        internal static bool HasErrorFor(this ParseResult parseResult, Option option)
+        internal static bool HasErrorFor(this ParseResult parseResult, Option option, [NotNullWhen(true)] out ParseError? error)
```

Comment:
this is also a perf optimization: previously we were iterating the errors up to three times, now it's only once and the caller does not need to search for the error again